src/GafferSceneUIModule/LightEditorBinding.cpp

```diff
@@ -432,15 +432,24 @@ class SetMembershipColumn : public InspectorColumn
 
 		CellData headerData( const IECore::Canceller *canceller ) const override
 		{
-			return InspectorColumn::headerData( canceller );
+			CellData result = InspectorColumn::headerData( canceller );
+
+			ConstPathMatcherDataPtr setMembersData = m_scene->set( m_setName );
```

Comment:
Since we're going to be setting up our own context for this call, would it be slightly more efficient to use `ScenePlug::SetScope` and `m_scene->setPlug()->getValue()`? I don't expect it would make much difference, but I thought of that while deciding what type of scope to use.

Reply:
It would be a bit quicker, but probably not in a way that matters for a one-off query for the header value...

Reply:
Fixed in https://github.com/GafferHQ/gaffer/pull/5108/commits/2ec5513ef04644db338bd1ddaa8354fd0aa15433. I'm not checking that `sceneInput->ancestor<ScriptNode>()` returns a non-null value, is there a way it could not have a `ScriptNode` ancestor?

Reply:
Since the LightEditor derives from Editor and Editor requires there to be a ScriptNode, I think that is pretty much impossible at the moment. But because these column types might eventually become part of the public API, I suppose there's a small glimmer of possibility. If we were worried, we could express it this way perhaps :

```
auto scriptNode = m_scene->source()->ancestor<ScriptNode>();
Context::Scope contextScope( scriptNode ? scriptNode->context() : nullptr );
ConstPathMatcherDataPtr setMembersData = m_scene->set( m_setName );
```

`source()` is a nice alternative to `getInput()` because it never returns null, and `Context::Scope( nullptr )` is a no-op, which is also convenient in this case.